lib/value/src/kind/collection.rs

```diff
@@ -136,18 +148,14 @@ impl<T: Ord> Collection<T> {
             .known
             .values_mut()
             .reduce(|lhs, rhs| {
-                lhs.merge_keep(rhs.clone(), true);
+                lhs.merge_keep(rhs.clone(), false);
```

Comment:
Yes, I think this one got lost. I'll add an issue with more info, but as far as I can tell this has always been broken (at least since 0.19.0) and was refactored a few times keeping the bug. I'll link the bug soon with an example if you are more curious, and add a test.

Reply:
https://github.com/vectordotdev/vector/issues/13685